learning_resources/views.py

```diff
@@ -376,6 +378,125 @@ class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ReadOnlyModelViewSet
+)
+class LearningResourceListRelationshipViewSet(viewsets.GenericViewSet):
+    """
+    Viewset for managing relationships between Learning Resources
+    and User Lists / Learning Paths
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    filter_backends = [MultipleOptionsFilterBackend]
+    queryset = LearningResourceRelationship.objects.select_related("parent", "child")
+    http_method_names = ["patch"]
+
+    def get_serializer_class(self):
+        if self.action == "userlists":
+            return UserListRelationshipSerializer
+        elif self.action == "learning_paths":
+            return LearningResourceRelationshipSerializer
+        return super().get_serializer_class()
+
+    @extend_schema(
+        summary="Set User List Relationships",
+        description="Set User List Relationships on a given Learning Resource.",
+        responses={200: UserListRelationshipSerializer(many=True)},
+    )
+    @action(detail=True, methods=["patch"], name="Set User List Relationships")
+    def userlists(self, request, *args, **kwargs):  # noqa: ARG002
+        """
+        Set User List relationships for a given Learning Resource
+        """
+        learning_resource_id = kwargs.get("pk")
+        user_list_ids = request.query_params.getlist("userlist_id")
+        current_relationships = UserListRelationship.objects.filter(
+            parent__author=request.user, child_id=learning_resource_id
+        )
+        current_relationships.exclude(parent_id__in=user_list_ids).delete()
+        for userlist_id in user_list_ids:
+            last_index = 0
+            for index, relationship in enumerate(
+                UserListRelationship.objects.filter(
+                    parent__author=request.user, parent__id=userlist_id
+                ).order_by("position")
+            ):
+                relationship.position = index
+                relationship.save()
+                last_index = index
+            UserListRelationship.objects.create(
+                parent_id=userlist_id,
+                child_id=learning_resource_id,
+                position=last_index + 1,
+            )
+        serializer = UserListRelationshipSerializer(current_relationships, many=True)
+        return Response(serializer.data)
+
+    @extend_schema(
+        summary="Set Learning Path Relationships",
+        description="Set Learning Path Relationships on a given Learning Resource.",
+        responses={200: LearningResourceRelationshipSerializer(many=True)},
+    )
+    @action(detail=True, methods=["patch"], name="Set Learning Path Relationships")
+    def learning_paths(self, request, *args, **kwargs):  # noqa: ARG002
+        """
+        Set Learning Path relationships for a given Learning Resource
+        """
+        if not is_learning_path_editor(request):
+            return Response(
+                {"error": "You do not have permission to edit learning paths"},
+                status=403,
+            )
```

Comment:
I'm fairly certain you can pass permission_classes to an action decorator, so you could pass `permission_classes=[HasLearningPathPermissions]` to the decorator, but this is good too.